porcupine/plugins/pluginmanager.py

```diff
@@ -12,6 +12,8 @@
 
 log = logging.getLogger(__name__)
 
+dialog = None
```

Comment:
You need to tell mypy that this is not always None, even though the initial assignment sets it to None:

```suggestion
dialog: Optional[tkinter.Toplevel] = None
```

This hides other bugs that we'll find with mypy once this suggestion is applied. I created https://github.com/python/mypy/issues/10634.